lib/services/resource.go

```diff
@@ -254,9 +254,14 @@ type MarshalConfig struct {
 	// Version specifies particular version we should marshal resources with
 	Version string
 
-	// SkipValidation is used to skip schema validation.
+	// SkipValidation turns off schema validation for marshalers that
+	// validate by default.
 	SkipValidation bool
 
+	// DoValidation turns on schema validation for marshalers that
+	// don't validate by default.
+	DoValidation bool
```

Comment:
This is confusing to me, we have two variables (four choices) for something that only has two choices. If validation needs to be controllable, just have on by default but pass SkipValidation where need in the client?